src/thread_task.rs

```diff
@@ -197,15 +197,19 @@ impl Rav1dTasks {
             let new_start = self.len();
             {
                 let mut pending_tasks = self.pending_tasks.lock();
+                if pending_tasks.len() == 0 {
+                    // A spurious merge.
+                    // We don't synchronise the two parts of `add_pending` so this is always possible.
+                    return false;
+                }
                 let mut tasks = self.tasks.try_write().unwrap();
                 tasks.extend(pending_tasks.drain(..));
             }
             {
                 // take a read lock because reset_task_cur takes a read lock also
                 let tasks = self.tasks.try_read().unwrap();
-                for t in tasks[new_start..].iter() {
-                    self.insert_tasks(c, t.frame_idx, 0);
-                }
+                let frame_idx = tasks[new_start].frame_idx;
+                self.insert_tasks(c, frame_idx, 0);
```

Comment:
What's the reason for why we can do this?

Reply:
Okay, that makes sense.

Reply:
See new ce1f0a42cfa51065194292dda8a7d85ba9d448d0